dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -276,6 +276,10 @@ private ListenableFuture<Response> handleThrowable(Throwable clientSideThrowable
                 if (requestCanBeRetried() && shouldAttemptToRetry(clientSideThrowable)) {
                     callsiteStacktrace.ifPresent(clientSideThrowable::addSuppressed);
                     Meter retryReason = retryDueToThrowable.apply(clientSideThrowable);
+                    // Retry immediately when we failed to reach the server, since it wasn't a capacity issue
+                    // Note that this should retry on another node with DialogueNodeSelectionStrategy.BALANCED,
```

Comment:
"should" feels like too strong of language here. There is no guarantee that we will retry on another node.

I would phrase this like "It is expected that most of the time the request will be retried on another node..."